vm/src/obj/objtype.rs

```diff
@@ -494,9 +516,31 @@ pub fn new(
     _base: PyClassRef,
     bases: Vec<PyClassRef>,
     dict: HashMap<String, PyObjectRef>,
+    optional_vm: Option<&VirtualMachine>,
 ) -> PyResult<PyClassRef> {
+    // Check for duplicates in bases.
+    let mut unique_bases = HashSet::new();
+    for base in bases.iter() {
```

Comment:
`bases.iter().all(move |x| unique_bases.insert(x))`

Reply:
I need the failing class for the error string (so it matches CPython). I could do this twice, first to check if there are duplicates and the second time to find it, but that sounds worse.